Automattic/jetpack
# Boost: Use Blog ID in links to WPCOM instead of site slug

## Proposed changes:
Adjust checkout links in Boost plugin to use blog ID instead of slug for multi-URL support.

### Other information:


## Jetpack product discussion
pf5801-93-p2

## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:
To test the blog, you'll need two domains pointing to the same site. One way to do that is to setup two proxying tunnels to the same local environment.

**Connect Jetpack with a free plan on URL A, and load it via URL B.**

**To also fully test the Calypso changes, please test the changes separately on site-only connected site.**

If your site falls into IDC due to concurrency issues with IDC secret validation (fix pending), remove `jetpack_sync_error_idc` option.

1. Using URL B (not the main cache site URL), go to "Jetpack -> Boost" and click "Get Boost".
2. Confirm you were taken to checkout, and the page URL contains URL A (main cache site URL).
3. Don't finish the checkout, go back and choose the free plan. You should see the Boost dashboard.
4. Scroll down to the "Upgrade" link and click it. You should see the upgrade upsell card, click "Upgrade Jetpack Boost".
5. Confirm you were taken to checkout, and the page URL contains URL A (main cache site URL). 
6. Finish the checkout. Confirm that you're back in the Boost Dashboard and the plan was upgraded.
